### DISCUSSION TITLE
Alda on Raspberry Pi

### DISCUSSION THREAD

**Discussion by daveyarwood** [alda-lang/alda#456] [Ideas]
Continuing from conversations in [the Alda Slack group](https://slack.alda.io), tracking the work to be done here:

* [x] Add ARM builds to `client/bin/build`
* [x] Tune JVM options so that alda-player processes use less memory?
* [x] Add arm64 release testing to CircleCI build (looks like they don't support 32-bit ARM, so can't test ARM 7)
* [x] Update Alda API to include arm-7 and arm64 releases
* [x] Ensure that `alda update` can find the right releases for ARM 7, arm64
* [x] Ensure that ARM releases are included at alda.io/install

**Comment by Arthurmeade12** [comment]
As of June 2025 (Edit: as of 688a915), I was able to build a working alda install for my arm64 Fedora Desktop with no changes to the code, except adding `build linux arm64` to the end of client/bin/build . So, to anyone else out there with an arm64 linux machine - and some knowledge of how to manually build and install cli projects - this is working. 
Proof: https://asciinema.org/a/ykArg3dFCR86itlsV5nacKvBQ (this will expire on 6/30/2025)

@daveyarwood could you be persuaded to add this to the official builds on alda.io? Thanks for making a great project!

**Reply by daveyarwood** [reply]
That's great! Yes, I was already planning to do that eventually. I even made [this branch](https://github.com/alda-lang/alda/compare/master...arm-build) back in 2021 that included similar changes to the build script.

I don't really do anything with Raspberry Pi myself (although I think they're cool), but it's nice to know that there is still interest in getting Alda to run on it!

I'll try to find some time soon to test that branch and merge it, and then work on getting ARM builds included in the official builds on the website.

**Reply by daveyarwood** [reply]
@Arthurmeade12 I spent some time tonight working on support for ARMv7 and ARM64. Releases are now available on the [install page](https://alda.io/install/):

<img width="768" height="426" alt="2025-09-19-220037_768x426_scrot" src="https://github.com/user-attachments/assets/36181a17-0412-4273-b8b7-8bd7bb96b63e" />

I don't have any Linux ARM devices to test this out, but if you're able to test, that would be great!

I did set up our CI build to test in an Linux ARM64 container, and that's looking good. I have no idea if the sound works, though :smile:

**Reply by Arthurmeade12** [reply]
That's great to hear!

The successful test I did was actually on an Apple M2 Mac mini (running Linux) with a ARMv8.6-A processor, but I am happy to see if these work all the same when I get home.
Thanks for the work on Alda! It's a great project 👍

**Reply by Arthurmeade12** [reply]
Slightly unrelated, but I maintain the Arch Linux AUR packages for Alda and am wondering if there is a mailing list or update notification when Alda releases (as it did yesterday), so I can update the package quickly. Thanks again!

**Reply by daveyarwood** [reply]
Thanks for maintaining the AUR packages!

There isn't a mailing list, but one thing I always do (it's part of the automated release process in CI) is post in the #general channel in [Alda Slack](https://slack.alda.io). You could perhaps set up a notification in Slack?
